lib/ansible/modules/windows/win_blockinfile.ps1

```diff
@@ -0,0 +1,330 @@
+# and Yaegashi Takeshi's (@yaegashi) blockinfile
+
+function WriteLines($outlines, $path, $linesep, $encodingobj, $validate, $check_mode) {
+
+    $acl = $null
+
+    Try {
+        If (Test-Path -LiteralPath $path) {
+            $acl = Get-Acl -Path $path
+        }
+    }
+    Catch {
+        Fail-Json @{} "Cannot get ACL from original file temporary file! ($($_.Exception.Message))";
+    }
+    Try {
+        $temppath = [System.IO.Path]::GetTempFileName();
+    }
+    Catch {
+        Fail-Json @{} "Cannot create temporary file! ($($_.Exception.Message))";
+    }
+    $joined = $outlines -join $linesep;
+    [System.IO.File]::WriteAllText($temppath, $joined, $encodingobj);
+
+    If ($validate) {
+
+        If (-not ($validate -like "*%s*")) {
+            Fail-Json @{} "validate must contain %s: $validate";
+        }
+
+        $validate = $validate.Replace("%s", $temppath);
+
+        $parts = [System.Collections.ArrayList] $validate.Split(" ");
+        $cmdname = $parts[0];
+
+        $cmdargs = $validate.Substring($cmdname.Length + 1);
+
+        $process = [Diagnostics.Process]::Start($cmdname, $cmdargs);
+        $process.WaitForExit();
+
+        If ($process.ExitCode -ne 0) {
+            [string] $output = $process.StandardOutput.ReadToEnd();
+            [string] $error = $process.StandardError.ReadToEnd();
+            Remove-Item $temppath -force;
+            Fail-Json @{} "failed to validate $cmdname $cmdargs with error: $output $error";
+        }
+
+    }
+
+    # Commit changes to the path
+    $cleanpath = $path.Replace("/", "\");
+    Try {
+        Copy-Item $temppath $cleanpath -force -ErrorAction Stop -WhatIf:$check_mode;
+    }
+    Catch {
+        Fail-Json @{} "Cannot write to: $cleanpath ($($_.Exception.Message))";
+    }
+
+    If ($acl) {
+        Try {
+            Set-Acl -Path $cleanpath -AclObject $acl -WhatIf:$check_mode 
+        }
+        Catch {
+            Fail-Json @{} "Cannot set ACL on new file! ($($_.Exception.Message))";
+        }
+    }
+
+    Try {
+        Remove-Item $temppath -force -ErrorAction Stop;
+    }
+    Catch {
+        Fail-Json @{} "Cannot remove temporary file: $temppath ($($_.Exception.Message))";
+    }
+
+    return $joined;
+}
+
+
+# Backup the file specified with a date/time filename
+function BackupFile($path, $check_mode) {
+    $backuppath = $path + "." + [DateTime]::Now.ToString("yyyyMMdd-HHmmss");
+    Try {
+        Copy-Item $path $backuppath -WhatIf:$check_mode;
+    }
+    Catch {
+        Fail-Json @{} "Cannot copy backup file! ($($_.Exception.Message))";
+    }
+    If (-not $check_mode) {
+        Try {
+            Get-Acl -Path $path | Set-Acl -Path $backuppath
+        }
+        Catch {
+            Fail-Json @{} "Cannot copy ACL to backup file! ($($_.Exception.Message))";
+        }
+    }
+    return $backuppath;
+}
+
+# Parse the parameters file dropped by the Ansible machinery
+$params = Parse-Args $args -supports_check_mode $true;
```

Comment:
I noticed this module is still using the old Powershell module interface.
Would you care to look at the new Powershell module interface (look at some of the other files) and rewrite this using it ?

Reply:
I actually noticed that in some of the modules (I made a note of it in my description), but the docs have this method as the example so went this route. I'll switch it over tomorrow...

Reply:
Yeah, we should update the docs, but this is v2.8 material so unreleased as such :-)